README.md

```diff
@@ -1,2 +1,58 @@
 # koa-npm-resolution
-Koa middleware that transforms NPM package specifiers to relative paths
+
+This is a middleware for Koa servers that resolves NPM package specifiers in ES modules to relative paths for use on the web.  It is currently only intended for use as middleware in development servers to fascilitate build-free testing/iteration.
+
+## What?
+
+So you want to write your front-end code using ES modules and you're using NPM packages and you don't want to run a build step to transform those various import statements from convenient forms like: `import stuff from 'stuff'` into `import stuff from './node_modules/stuff/index.js'` etc.  This middleware simply transforms them when serving up the response from your server.
+
+That means you can use the middleware in a simple server that delivers static files as well as in a proxy server that might be piping files from a test server such as the one `karma` starts up.  (See [karma testing setup](#karma-testing-setup) below.)
+
+## Installation
+
+```sh
+$ npm install --save koa-npm-resolution
+```
+
+## Usage
+
+Create your own mini-development server.  This one depends on `koa` and `koa-static`, so you'll need to `npm install --save-dev koa koa-static` for your project to use it.
+
+```js
+/* dev-server.js */
+const Koa = require('koa')
```

Comment:
semicolons!

Reply:
I'd been using no semi-colons ala "Standard" for my handcrafted pure JavaScript lately and semi-colons for TypeScript since with clang-format and that habit leaked into my example code.  I'll remember not do that for stuff like this (though I *want* to.)